src/compiler/watchPublic.ts

```diff
@@ -273,6 +273,7 @@ namespace ts {
         let sharedExtendedConfigFileWatchers: ESMap<Path, SharedExtendedConfigFileWatcher<Path>>; // Map of file watchers for extended files, shared between different referenced projects
         let extendedConfigCache = host.extendedConfigCache;                 // Cache for extended config evaluation
         let changesAffectResolution = false;                                // Flag for indicating non-config changes affect module resolution
+        let reportWatchDiagnosticOnCreateProgram: DiagnosticMessage | undefined; // Diagnostic message to report in synchronizeProgram before new program creation

```

Comment:
Personally, I would find a bool more readable - this creates the impression that multiple distinct messages are possible.

Reply:
Switched to a bool.